### ISSUE TITLE
Reading a file fails if some predicates are not present

### ISSUE THREAD

**Issue by theWebalyst** [jeff-zucker/solid-file-client#30]
I encountered an error loading a blog post (actually using [solid-filemanager](https://github.com/Otto-AA/solid-filemanager/) which incorporates some of your code, so I'm reporting it here first). The test was conducted on SAFE Network, reading a blog post originally created by Solid Plume, but it is a general issue, not just related to this specific example.

The code failed in [folderUtils.js getStats()](https://github.com/jeff-zucker/solid-file-client/blob/master/src/folderUtils.js#L26-L30) because the `mtime` value was not present, but as I understand LDP we shouldn't assume any particular predicates are present for any `rdf:type`:
```
  return {
    modified: modified,
    size: graph.any(subjectNode, size, undefined).value,
    mtime: graph.any(subjectNode, mtime, undefined).value,
  };
```
I'm guessing NSS will probably insert some attributes (e.g. add `mtime` where `modified` is present) but that is not something we can rely on if Solid is to use different storage backends and server implementations.

As I understand the LDP spec, for an LDP-RS we can only assume the presence of an `rdf:type`, should not  assume any specific predicates will be present for a given type, nor that the same predicates will be available in different resources of the same type!

So as I understand it, we should write code that checks all values before using them. See: 4.3.1 of https://www.w3.org/TR/ldp/

I'm new at RDF, so I could be totally wrong! :smile:

**Comment by jeff-zucker** [comment]
Good catch, thanks!  I guess I will trap the error so if the predicate doesn't exist it returns an undefined value for that predicate but the method succeeds.  I should also change the documentation to state that values for file stats are not guaranteed to be available on all servers.

**Comment by Otto-AA** [comment]
A follow up question from me: Is there a particular reason, why the getStats function returns `if(typeof(modified)==="undefined") return false;`? My app doesn't make use of this modification date, so I could scrap it if if there is no deeper meaning behind it.

**Comment by jeff-zucker** [comment]
I was noticing that too.  I'll change it to also return true but with an undefined modified.

**Comment by jeff-zucker** [comment]
Fixed in version 1.0.0.
